### PR TITLE
Introduce code node & get rid of code map

### PR DESCRIPTION
**Pull request by mandrigin**

### CODE REVIEW

**Comment by mandrigin** [comment]
bug:
```
Processed 483050 blocks (2698.5977653631285 blocks/sec)
Wrong block hash 00e9a59816f8fd52a28206acced33bc832df7d40fe2b7ce50cae7962ee63c95e in block 483055
stoppped at block number: 483055
```

**Comment by mandrigin** [comment]
sync test is running here: https://ssh.cloud.google.com/projects/torquem/zones/us-central1-a/instances/code-node-test?authuser=2&hl=en_US&projectNumber=933467666552

**Comment by mandrigin** [comment]
also:
```
--- FAIL: TestBlockchain (0.01s)
    --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json (0.01s)
        --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json/extCodeHashOfDeletedAccountDynamic_Constantinople (0.00s)
            block_test.go:52: block #1 insertion into chain failed: account not found with key: 0786a8869ff4044202667cbe946592cc381d99ec19b0ad50e1f77363bc8cd48c: db: key not found
        --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json/extCodeHashOfDeletedAccountDynamic_ConstantinopleFix (0.00s)
            block_test.go:52: block #1 insertion into chain failed: account not found with key: 0786a8869ff4044202667cbe946592cc381d99ec19b0ad50e1f77363bc8cd48c: db: key not found
        --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json/extCodeHashOfDeletedAccountDynamic_Istanbul (0.00s)
            block_test.go:52: block #1 insertion into chain failed: account not found with key: 0786a8869ff4044202667cbe946592cc381d99ec19b0ad50e1f77363bc8cd48c: db: key not found
```

**Comment by mandrigin** [comment]
> also:
> 
> ```
> --- FAIL: TestBlockchain (0.01s)
>     --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json (0.01s)
>         --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json/extCodeHashOfDeletedAccountDynamic_Constantinople (0.00s)
>             block_test.go:52: block #1 insertion into chain failed: account not found with key: 0786a8869ff4044202667cbe946592cc381d99ec19b0ad50e1f77363bc8cd48c: db: key not found
>         --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json/extCodeHashOfDeletedAccountDynamic_ConstantinopleFix (0.00s)
>             block_test.go:52: block #1 insertion into chain failed: account not found with key: 0786a8869ff4044202667cbe946592cc381d99ec19b0ad50e1f77363bc8cd48c: db: key not found
>         --- FAIL: TestBlockchain/ValidBlocks/bcStateTests/extCodeHashOfDeletedAccountDynamic.json/extCodeHashOfDeletedAccountDynamic_Istanbul (0.00s)
>             block_test.go:52: block #1 insertion into chain failed: account not found with key: 0786a8869ff4044202667cbe946592cc381d99ec19b0ad50e1f77363bc8cd48c: db: key not found
> ```

fixed

**Comment by mandrigin** [comment]
> bug (stateless):
> 
> ```
> Processed 483050 blocks (2698.5977653631285 blocks/sec)
> Wrong block hash 00e9a59816f8fd52a28206acced33bc832df7d40fe2b7ce50cae7962ee63c95e in block 483055
> stoppped at block number: 483055
> ```

fixed too, ready to be merged from my perspective

**Comment by mandrigin** [comment]
Stateless test: processed **1185044** blocks successfully. I don't have more in my local chaindata at the moment.

File: core/state/database.go [hunk 1]
```diff
@@ -183,14 +192,6 @@ type TrieDbState struct {
 }
 
 func NewTrieDbState(root common.Hash, db ethdb.Database, blockNr uint64) (*TrieDbState, error) {
```
Comment (AlexeyAkhunov) [inline core/state/database.go]: Now there is no reason to have the second return value (error). Can clean it up in a separate PR

**Pull request merged**